src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -141,39 +141,52 @@ private async Task InvokeWorkerAsync(IProgressTracker progressTracker, Cancellat
         {
             await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            IEnumerable<CodeActionOperation> operations = null;
+            // Todo: do we want to measure the amount of time it takes get the options from the user?
+            object options = null;
             if (CodeAction is CodeActionWithOptions actionWithOptions)
             {
-                var options = actionWithOptions.GetOptions(cancellationToken);
-                if (options != null)
-                    operations = await GetOperationsAsync(actionWithOptions, options, cancellationToken).ConfigureAwait(true);
+                options = actionWithOptions.GetOptions(cancellationToken);
             }
-            else
-            {
-                operations = await GetOperationsAsync(progressTracker, cancellationToken).ConfigureAwait(true);
-            }
-
-            AssertIsForeground();
 
-            if (operations != null)
+            using (Logger.LogBlock(
+                   FunctionId.CodeFixes_GetAndApplyChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
             {
-                // Clear the progress we showed while computing the action.
-                // We'll now show progress as we apply the action.
-                progressTracker.Clear();
-
+                IEnumerable<CodeActionOperation> operations = null;
                 using (Logger.LogBlock(
-                    FunctionId.CodeFixes_ApplyChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
+                       FunctionId.CodeFixes_GetChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
+                {
+                    if (options != null && CodeAction is CodeActionWithOptions codeActionWithOptions)
```

Comment:
don't need the `is` check, should be a blind cast as that's the only way to get options not being null.